atlp-rwanda/cabal-phantom-backend
# #G7j46apS-FT(Backend)Assign driver to buses

### **What does this PR do?**
This PR is for assigning driver to buses

### **Description of Task to be completed?**

- Operator should be able to assign and unassign driver to a bus
   - **PATCH /api/v1/buses/assigndriver/:busId**
   - **PATCH /api/v1/buses/unassigndriver/:busId**


- The driver should receive a bus assignment and unassignment email & in-app notifications.

- The operator should be able to view a paginated list of driver to bus assignments
   - **GET /api/v1/buses/assignedbuses?page={page}&limit={limit}**

### **How should this be manually tested?**

- .You can use [Swagger documentation](https://phantom-caba-ft-assign--ljkwgj.herokuapp.com/api-docs) and follow the instructions below

- By using postman, use this url: https://phantom-caba-ft-assign--ljkwgj.herokuapp.com **(Optional)**

- Login as admin {"email": "**admin@phantom.com**","password":"**admin**"}

- Register new user and assign him/her a role of _driver_

  - Register new user: **POST /api/v1/auth/register**
  - Assign role: **PATCH /api/v1/auth/updateuser**

- Make assignment and unassignment of driver to bus with the endpoint specified above. For each action check the email notification.

**What are the relevant pivotal tracker stories?**
[#G7j46apS](https://trello.com/c/G7j46apS/12-assign-drivers-to-buses)

**Screenshots (if appropriate)**
<img width="960" alt="tst1" src="https://user-images.githubusercontent.com/68014898/97796503-63dc2a80-1c1b-11eb-8f45-c37961c25e07.png">
<img width="960" alt="tst2" src="https://user-images.githubusercontent.com/68014898/97796510-7191b000-1c1b-11eb-91e0-51b71df3adb5.png">
<img width="960" alt="tst3" src="https://user-images.githubusercontent.com/68014898/97796514-7a828180-1c1b-11eb-9c09-5204b3ee7ce5.png">
